server_sinks_test.go

```diff
@@ -173,23 +173,3 @@ func TestNewDatadogMetricSinkConfig(t *testing.T) {
 	// Verify that the values got set	assert.Equal(t, "apikey", sink.APIKey)
 	assert.Equal(t, "http://api", sink.DDHostname)
 }
-
-func TestNewPrometheusMetricSinkConfig(t *testing.T) {
```

Comment:
You've removed this test without replacing it with an equivalent elsewhere (avoiding an import cycle?); maybe it could move into the Prometheus sink's test suite?

Reply:
Would the `TestMetricSinkName` and `TestParseConfig` tests effectively capture everything this test is doing?

Reply:
These feel more like they're testing the integration point that used to live in `server.go`; the point of this test might have been to test server spin-up from a general config? To make sure the specific config type would get picked up and properly placed into a new `Server`.

The closest analog seems to be in `cmd/veneur/main.go`, starting around line 57, but that doesn't look designed to be tested right now.

Reply:
I was writing the comment below to explain why the other tests I added are an equivalent replacement for this test but thinking about it some more, that may not be true. While both test that the sink name is properly configured, server_sinks_test checks that using NewFromConfig while my other test uses CreateMetricSink.

Leaving it below for context.
---

Ah yes I can see why it looks that way. Removing it did help fix an import cycle but I rewrote it using the newly added "CreateMetricSink" function as seen in TestMetricSinkName. I can't recall for sure why I rewrote it but I think it's because metricSinks (L193) is a private field and I couldn't access it when I moved it inside the prometheus package.

TestCreateMetricSink covers the rest of the check.

Reply:
Per offline discussion with @jcrpaquin-stripe, we'll do some refactoring in a follow up PR and write a new NewFromConfig test that should cover Prometheus, Datadog and the other sinks. Meanwhile, I'll go ahead and merge this PR.